### PR TITLE
DOC: add docstring to client methods

### PR DESCRIPTION
**Pull request by RayJi01**
ATT

### CODE REVIEW

File: xinference/client.py [hunk 1]
```diff
@@ -58,10 +58,42 @@ def generate(
             Union["LlamaCppGenerateConfig", "PytorchGenerateConfig"]
         ] = None,
     ) -> Union["Completion", Iterator["CompletionChunk"]]:
+        """
+        Function to call the generate functionality of model on the server side.
```
Comment (UranusSeven) [inline xinference/client.py:62]: Creates a completion for the provided prompt and parameters.

File: xinference/client.py [hunk 2]
```diff
@@ -58,10 +58,42 @@ def generate(
...
+
+        Parameters
+        ----------
+        prompt : str
+            The user's message and the user's input
```
Comment (UranusSeven) [inline xinference/client.py:67]:
```suggestion
            The user's input.
```

File: xinference/client.py [hunk 3]
```diff
@@ -58,10 +58,42 @@ def generate(
...
+        generate_config: Optional["ChatglmCppGenerateConfig"]
+            Additional Configuration for the Model text generation.
```
Comment (UranusSeven) [inline xinference/client.py:69]:
```suggestion
            Additional configurations for completion.
```

File: xinference/client.py [hunk 4]
```diff
@@ -58,10 +58,42 @@ def generate(
...
+
+        Returns
+        -------
+        Union["Completion", Iterator["CompletionChunk"]]
+            The completion chunk generated by the Model.
```
Comment (UranusSeven) [inline xinference/client.py:74]: `stream=True` -> Iterator["CompletionChunk"]
`stream=False` -> "Completion"

File: xinference/client.py (hunk 2 above)
Comment (RayJi01) [reply xinference/client.py]: resolved in latest version

File: xinference/client.py (hunk 1 above)
Comment (RayJi01) [reply xinference/client.py]: resolved in the latest version

File: xinference/client.py (hunk 3 above)
Comment (RayJi01) [reply xinference/client.py]: resolved in the latest version

File: xinference/client.py (hunk 4 above)
Comment (RayJi01) [reply xinference/client.py]: resolved in the latest version

File: xinference/client.py [hunk 5]
```diff
@@ -76,6 +111,27 @@ def chat(
...
+        """
+        Given a list of messages comprising a conversation, the model will return a response.
+
+        Parameters
+        ----------
+        prompt : str
+            The user's input.
+        chat_history: Optional[List["ChatCompletionMessage"]]
+            A list of messages comprising the conversation so far.
+        generate_config: Optional["ChatglmCppGenerateConfig"]
```
Comment (UranusSeven) [inline xinference/client.py:123]: Incorrect type.

Also it is better to let users know the difference between LlamaCppGenerateConfig and PytorchGenerateConfig and which one to use.

File: xinference/client.py [hunk 6]
```diff
@@ -58,10 +58,45 @@ def generate(
...
         ] = None,
     ) -> Union["Completion", Iterator["CompletionChunk"]]:
+        """
+        Creates a completion for the provided prompt and parameters.
+
+        Parameters
+        ----------
+        prompt : str
+            The user's input.
+        generate_config: Optional["ChatglmCppGenerateConfig"]
```
Comment (UranusSeven) [inline xinference/client.py:68]: Ditto.

File: xinference/client.py [hunk 7]
```diff
@@ -58,10 +58,45 @@ def generate(
...
+            Additional configurations for completion.
+
+        Returns
+        -------
+        Union["Completion", Iterator["CompletionChunk"]]
+            The Completion chunk of the model.
+            stream=True -> Iterator["CompletionChunk"]
```
Comment (UranusSeven) [inline xinference/client.py:75]: Explain `stream`.

File: xinference/client.py (hunk 6 above)
Comment (RayJi01) [reply xinference/client.py]: changed in the latest version

File: xinference/client.py (hunk 5 above)
Comment (RayJi01) [reply xinference/client.py]: changed the incorrect type

And add model specified instructions

File: xinference/client.py (hunk 7 above)
Comment (RayJi01) [reply xinference/client.py:81]: added explanation for both what happening to stream = true / stream = false in the latest version

File: xinference/client.py [hunk 8]
```diff
@@ -133,6 +254,34 @@ def generate(
             Union["LlamaCppGenerateConfig", "PytorchGenerateConfig"]
         ] = None,
     ) -> Union["Completion", Iterator["CompletionChunk"]]:
+        """
+        Creates a completion for the provided prompt and parameters via RESTful apis.
```
Comment (UranusSeven) [inline xinference/client.py:258]:
```suggestion
        Creates a completion for the provided prompt and parameters via RESTful APIs.
```

File: xinference/client.py [hunk 9]
```diff
@@ -270,7 +497,30 @@ def launch_model(
         quantization: Optional[str] = None,
         **kwargs,
     ) -> str:
-        model_uid = self.gen_model_uid()
+        """
+        Launch the Model based on the parameters we demanded on the server.
```
Comment (UranusSeven) [inline xinference/client.py:501]:
```suggestion
        Launch the model based on the parameters.
```

File: xinference/client.py [hunk 10]
```diff
@@ -270,7 +497,30 @@ def launch_model(
...
+
+        Parameters
+        ----------
+        model_name : str
+            The name of model we want. e.g. Vicuna.
+        model_size_in_billions : Optional[int]
+            The size (in billions) of the model.
+        model_format: Optional[str]
+            The format of the model. e.g. GGML.
```
Comment (UranusSeven) [inline xinference/client.py:510]:
```suggestion
            The format of the model.
```

File: xinference/client.py [hunk 11]
```diff
@@ -285,14 +535,56 @@ def launch_model(
...
         return self._isolation.call(coro)
 
     def list_models(self) -> List[Tuple[str, "ModelSpec"]]:
+        """
+        Retrieve the Model Spec from the Server.
+
+        Returns
+        -------
+        response_data : Dict[str, Dict[str, Any]]
+            The collection of model spec with their names on the server.
```
Comment (UranusSeven) [inline xinference/client.py:562]: Ditto.

File: xinference/client.py [hunk 12]
```diff
@@ -311,11 +603,21 @@ def __init__(self, base_url):
...
 
     @classmethod
-    def gen_model_uid(cls) -> str:
+    def _gen_model_uid(cls) -> str:
         # generate a time-based uuid.
         return str(uuid.uuid1())
 
     def list_models(self) -> Dict[str, Dict[str, Any]]:
+        """
+        Retrieve the Model Spec from the Server.
```
Comment (UranusSeven) [inline xinference/client.py:612]: Ditto.

File: xinference/client.py [hunk 13]
```diff
@@ -335,9 +637,32 @@ def launch_model(
         quantization: Optional[str] = None,
         **kwargs,
     ) -> str:
+        """
+        Launch the Model based on the demanding parameters on the server via RESTful api.
```
Comment (UranusSeven) [inline xinference/client.py:641]: Ditto.

File: xinference/client.py [hunk 14]
```diff
@@ -378,6 +718,29 @@ def _get_supervisor_internal_address(self):
         return response_data
 
     def get_model(self, model_uid: str) -> RESTfulModelHandle:
+        """
+        Launch the Model based on the parameters on the server via RESTful api.
```
[the remark this answers is not in the document]
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

File: xinference/client.py [hunk 15]
```diff
@@ -335,9 +637,32 @@ def launch_model(
...
+        Launch the Model based on the demanding parameters on the server via RESTful api.
+
+        Parameters
+        ----------
+        model_name : str
+            The name of model. e.g. Vicuna.
+        model_size_in_billions : Optional[int]
+            The size (in billions) of the model.
+        model_format: Optional[str]
+            The format of the model. e.g. GGML.
```
[the remark this answers is not in the document]
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

File: xinference/client.py (hunk 13 above)
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

File: xinference/client.py (hunk 11 above)
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

File: xinference/client.py [hunk 16]
```diff
@@ -285,14 +535,56 @@ def launch_model(
...
+            The model will be terminated.
+
+        """
+
         coro = self._supervisor_ref.terminate_model(model_uid)
         return self._isolation.call(coro)
 
     def list_models(self) -> List[Tuple[str, "ModelSpec"]]:
+        """
+        Retrieve the Model Spec from the Server.
```
[the remark this answers is not in the document]
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

File: xinference/client.py (hunk 10 above)
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

File: xinference/client.py (hunk 9 above)
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

File: xinference/client.py (hunk 8 above)
Comment (RayJi01) [reply xinference/client.py]: solved in the latest version

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**